lib/improver/tests/utilities/test_solar.py

```diff
@@ -0,0 +1,128 @@
+        longitudes = np.array([0.0, 10.0, -10.0, 180.0, -179.0])
+        result = solar_hour_angle(longitudes, 10, 12.0)
+        expected_result = np.array([-1.7832741, 8.2167259, -11.7832741,
+                                    178.2167259, -180.7832741])
+        self.assertArrayAlmostEqual(result, expected_result)
+
+
+class Test_solar_elevation(IrisTest):
+    """Test Calculation of the Solar Elevation."""
+
+    def test_basic_solar_elevation(self):
+        """Test the solar elevation for a single point over several hours."""
+        expected_results = [-0.460611756793, 6.78261282655,
+                            1.37746106416, -6.75237871867]
+        for i, hour in enumerate([8.0, 9.0, 16.0, 17.0]):
+            result = solar_elevation(50.0, 0.0, 10, hour)
+            self.assertAlmostEqual(result, expected_results[i])
+
+    def test_basic_solar_elevation_array(self):
+        """Test the solar elevation for an array of lats and lons."""
+        latitudes = np.array([50.0, 50.0, 50.0])
+        longitudes = np.array([-5.0, 0.0, 5.0])
+        expected_array = np.array([-3.1423043, -0.46061176, 2.09728301])
+        result = solar_elevation(latitudes, longitudes, 10, 8.0)
+        self.assertArrayAlmostEqual(result, expected_array)
+
+
+class Test_daynight_terminator(IrisTest):
+
+    """Test DayNight terminator."""
+
+    def setUp(self):
+        """Set up the longitudes."""
+        self.longitudes = np.linspace(-180.0, 180.0, 21)
+
+    def test_basic_winter(self):
+        """Test we get the terminator in winter."""
+        result = daynight_terminator(self.longitudes, 10, 12.0)
+        expected_lats = np.array([-67.79151577, -66.97565648, -63.73454167,
+                                  -56.33476507, -39.67331783, -4.36090124,
+                                  34.38678745, 54.03238506, 62.69165892,
+                                  66.55543647, 67.79151577, 66.97565648,
+                                  63.73454167, 56.33476507, 39.67331783,
+                                  4.36090124, -34.38678745, -54.03238506,
+                                  -62.69165892, -66.55543647, -67.79151577])
+        self.assertArrayAlmostEqual(result, expected_lats)
+
+    def test_basic_spring(self):
+        """Test we get the terminator in spring."""
+        result = daynight_terminator(self.longitudes, 100, 0.0)
+        expected_lats = np.array([-82.7926115, -82.44008918, -81.15273499,
+                                  -77.9520655, -68.09955141, -2.64493534,
+                                  67.37718951, 77.7625883, 81.07852035,
+                                  82.41166928, 82.7926115, 82.44008918,
+                                  81.15273499, 77.95206547, 68.09955141,
+                                  2.64493534, -67.37718951, -77.7625883,
+                                  -81.07852035, -82.41166928, -82.7926115])
+        self.assertArrayAlmostEqual(result, expected_lats)
```

Comment:
Try to write a test for the precise time when the terminator is at +90 or -90 (near, but probably not quite the equinox).
Also add a test for invalid input values.